lib/libv4l-gst/gst-backend.c

```diff
@@ -402,51 +388,30 @@ get_peer_pad_template_caps(GstElement *elem, const gchar *pad_name)
 }
 
 static gboolean
-get_supported_video_format_out(GstElement *appsrc, struct fmts **out_fmts,
-			       gint *out_fmts_num, gboolean enable_hevc)
+get_supported_video_format_out(struct gst_backend_priv *priv, struct fmts **out_fmts,
+			       gint *out_fmts_num, gboolean use_appsrc)
```

Comment:
IMHO it would be better to split this function instead of adding `use_appsrc`, since the procedures are totally different between `use_appsrc` case & `!use_appsrc` case.

In addition, please care following things:

* It would be better to change the funtion name of `use_appsrc` case, like `is_video_format_out_valid()`.
   Because the actual role of this function in this case is to check whether the pipeline was correctly constructed according to the specified format by the client.
* It would be better to change the function name of `!use_appsrc` case, like `fill_supported_video_format_out()`.
  And it should be called at `parse_config_file()` instead of at `enum_fmt_ioctl()`, because it's already determined at there.